plugin/dnstap/io.go

```diff
@@ -92,8 +92,10 @@ func (d *dio) write(payload *tap.Dnstap) error {
 }
 
 func (d *dio) serve() {
-	timeout := time.After(d.flushTimeout)
+	timeout := time.NewTimer(d.flushTimeout)
+	defer timeout.Stop()
 	for {
+		timeout.Reset(d.flushTimeout)
```

Comment:
Should this `timeout.Reset` be placed under `case <-timeout.C` branch, so the code is logically identical to `time.After` version?

Reply:
I don’t think it makes a difference, since the only other case in the select will exit. Iow, the only time we get back to the top of the for loop is when timeout fires.

Reply:
There is another branch `payload := <-d.queue`, reset timeout on top results in longer flush interval.

It doesn't make too much difference though, messages are flushed out once buffer is full.